### PR TITLE
chore: refacto variables names and secret creation strategy

### PR DESCRIPTION
**Pull request by johnfolly**
# Refacto variables names and secret creation strategy

- Added full documentation to Cosmotech Platform Azure Prerequisite Terraform
- Added parameter to make app secret creation optionnal

### CODE REVIEW

File: azure/create-platform-prerequisites/.terraform.lock.hcl
Comment (enocquet) [inline azure/create-platform-prerequisites/.terraform.lock.hcl:1]: we dont version this files as they can contain secrets and others sensitives informations

File: azure/create-platform-prerequisites/terraform.tfstate.backup
Comment (enocquet) [inline azure/create-platform-prerequisites/terraform.tfstate.backup:1]: Same here. No versionning

File: azure/create-platform-prerequisites/terraform.tfstate
Comment (enocquet) [inline azure/create-platform-prerequisites/terraform.tfstate:1]: Same here

File: azure/create-platform-prerequisites/terraform.tfvars [hunk 1]
```diff
@@ -4,18 +4,16 @@
...
   owner_list = [ "" ]
   webapp_url = ""
   dns_zone_name = ""
   dns_zone_rg = ""
   dns_record = ""
   vnet_iprange = ""
-  resource_group = ""
-  
-  user_app_role = [ {
+  user_user_app_role = [ {
```
Comment (enocquet) [inline azure/create-platform-prerequisites/terraform.tfvars]: typo ?

File: azure/create-platform-prerequisites/azure.tf [hunk 2]
```diff
@@ -18,6 +20,7 @@ provider "azurerm" {
   client_id       = var.client_id
   client_secret   = var.client_secret
   tenant_id       = var.tenant_id
+  skip_provider_registration = true
```
Comment (enocquet) [inline azure/create-platform-prerequisites/azure.tf]: what is the purpose of this parameter ?

**Review by enocquet — COMMENTED** [review]
you can format your code by doing `terraform fmt` to help maintain clarity inside repo

File: azure/create-platform-prerequisites/terraform.tfvars (hunk 1 above)
Comment (johnfolly) [reply azure/create-platform-prerequisites/terraform.tfvars]: Yes thanks

File: azure/create-platform-prerequisites/terraform.tfstate
Comment (johnfolly) [reply azure/create-platform-prerequisites/terraform.tfstate:1]: Ok

File: azure/create-platform-prerequisites/azure.tf (hunk 2 above)
Comment (johnfolly) [reply azure/create-platform-prerequisites/azure.tf]: It is to ensure that azure will not register it own providers 
<https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/resource_provider_registration>

Comment (enocquet) [reply azure/create-platform-prerequisites/azure.tf]: I understand that but are we sure we want to handle on our side the provider resources we need? I mean, why not letting azurerm provider handle this ?

File: azure/create-platform-prerequisites/variables.tf [hunk 3]
```diff
@@ -40,22 +38,24 @@ variable "stage" {
...
       "IA",
       "EA",
+      "Doc",
+      "Support",
       "Demo",
-      "Prod"
-    ], var.stage)
-    error_message = "Stage must be either: OnBoarding, Dev, QA, IA, EA, Demo, Prod."
+      "Prod",
+      "Uat"
```
Comment (enocquet) [inline azure/create-platform-prerequisites/variables.tf]: what does 'uat' mean ?

Comment (johnfolly) [reply azure/create-platform-prerequisites/variables.tf]: User Acceptance but i'll update the name to preprod

**Review by enocquet — APPROVED** [review]

File: azure/create-platform-prerequisites/variables.tf [hunk 4]
```diff
@@ -40,22 +38,24 @@ variable "stage" {
...
+      "Doc",
+      "Support",
       "Demo",
-      "Prod"
-    ], var.stage)
-    error_message = "Stage must be either: OnBoarding, Dev, QA, IA, EA, Demo, Prod."
+      "Prod",
+      "PreProd"
+    ], var.project_stage)
+    error_message = "Stage must be either: OnBoarding, Dev, QA, IA, EA, Demo, Prod, Uat."
```
Comment (enocquet) [inline azure/create-platform-prerequisites/variables.tf]: you surely forgot to swap 'uat' for 'preprod' ;)

**Review by enocquet — APPROVED** [review]
Dont forget the 'terraform fmt' and it's all good ^^ !

**Pull request merged**